app/src/main/java/io/intrepid/contest/screens/contestcreation/addcategoriestocontest/AddCategoriesFragment.java

```diff
@@ -0,0 +1,74 @@
+package io.intrepid.contest.screens.contestcreation.addcategoriestocontest;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import butterknife.OnTextChanged;
+import io.intrepid.contest.R;
+import io.intrepid.contest.base.BaseViewPagerFragment;
+import io.intrepid.contest.base.PresenterConfiguration;
+
+public class AddCategoriesFragment extends BaseViewPagerFragment<AddCategoriesPresenter> implements AddCategoriesContract.View {
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    protected int getLayoutResourceId() {
+        return R.layout.fragment_edit_contest_category;
+    }
+
+    @OnTextChanged(R.id.category_name_edittext)
+    public void onCategoryNamed(CharSequence newName, int start, int before, int count) {
+        onCategoryNameEntered(String.valueOf(newName));
+    }
+
+    @OnTextChanged(R.id.category_description_edittext)
+    public void onCategoryDescribed(CharSequence description, int start, int before, int count) {
+        onCategoryDescriptonEntered(String.valueOf(description));
+    }
+
+    @NonNull
+    @Override
+    public AddCategoriesPresenter createPresenter(PresenterConfiguration configuration) {
+        return new AddCategoriesPresenter(this, configuration);
+    }
+
+    @Override
+    public void onCategoryNameEntered(String name) {
+        presenter.editCategoryName(name);
+    }
+
+    @Override
+    public void onCategoryDescriptonEntered(String description) {
+        presenter.editCategoryDescription(description);
+    }
+
+    @Override
+    public void onCancelClicked() {
+        //todo
+    }
+
+    @Override
+    public void onEditComplete() {
+        submitText();
+    }
+
+    @Override
+    public boolean areAllFieldValid() {
+        return true;
```

Comment:
I'll allow this, but in general if we arent ready to use a feature fully, its better to save it for a second iteration of files. Rather then leave it in a state where the code isnt functioning it would be better to roll out the functionality when its ready.